### PR TITLE
docs: add CF tls termination

### PR DESCRIPTION
**Pull request by iamhopaul123**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: site/content/docs/developing/content-delivery.en.md [hunk 1]
```diff
@@ -53,4 +53,24 @@ With both of these setups, Copilot will provision CloudFront to use the [SSL/TLS
...
+## How do I use CloudFront to terminate TLS?
+
+You can optionally use CloudFront for TLS termination by configuring the env manifest as
+
+```yaml
+cdn:
+  tls_termination: true
+```
+
+And traffic from `CloudFront → Application Load Balancer (ALB) → ECS` will be HTTP only. This brings a very important benefit which is faster TLS termination, since the CloudFront edges are usually closer to the viewers.
```
Comment (efekarakus) [inline site/content/docs/developing/content-delivery.en.md]: nit:
```suggestion
And traffic from `CloudFront → Application Load Balancer (ALB) → ECS` will be HTTP only. This brings the benefit of terminating TLS at a geographically closer endpoint to the end user for faster TLS handshakes.
```

File: site/content/docs/developing/content-delivery.en.md [hunk 2]
```diff
@@ -53,4 +53,24 @@ With both of these setups, Copilot will provision CloudFront to use the [SSL/TLS
...
+
+However, if HTTPS is enabled by either having the app domain or importing any certificates in the environment, you must turn off ALB http redirect by updating the service manifests as
```
Comment (efekarakus) [inline site/content/docs/developing/content-delivery.en.md]: I wonder if we should use the `!!! attention ""` [admonition](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) here to highlight that they should do this first?

```
Before terminating TLS at the CDN level, we recommend:
1. First, disabling [HTTP to HTTPS redirection](fill-link-service-manifest) for your Load Balanced Web Services first. 
   ```yaml
   http:
      redirect: false
   ```
2. Run `copilot svc deploy`

Once all your services no longer, re-direct you can safely enable CloudFront TLS termination.
```

File: site/content/docs/manifest/lb-web-service.en.md [hunk 3]
```diff
@@ -259,6 +259,9 @@ http:
   hosted_zone: Z0873220N255IR3MTNR4
 # Also see http.alias array of maps example, above.
 ```
+<span class="parent-field">http.</span><a id="http-disable-redirect" href="#http-disable-redirect" class="field">`disable_redirect`</a> <span class="type">Boolean</span>
```
Comment (efekarakus) [inline site/content/docs/manifest/lb-web-service.en.md]: @dannyrandall @iamhopaul123 

Any reason why we preferred a double negation? Typically, we should avoid them as a code smell: https://refactoring.com/catalog/removeDoubleNegative.html

What do y'all think of changing the field to `http.redirect: false` and by default it's `true`.

File: site/content/docs/developing/content-delivery.en.md [hunk 4]
```diff
@@ -57,20 +57,17 @@ You can restrict incoming traffic to come from a certain source. For CloudFront,
 
 ## How do I use CloudFront to terminate TLS?
 
+!!! attention
+    1. First, disabling [HTTP to HTTPS redirection](../manifest/lb-web-service/#http-redirect-to-https) for your Load Balanced Web Services first.
```
Comment (efekarakus) [inline site/content/docs/developing/content-delivery.en.md]:
```suggestion
    1. First, disable [HTTP to HTTPS redirection](../manifest/lb-web-service/#http-redirect-to-https) for your Load Balanced Web Services.
```

File: site/content/docs/developing/content-delivery.en.md [hunk 5]
```diff
@@ -57,20 +57,17 @@ You can restrict incoming traffic to come from a certain source. For CloudFront,
...
+    2. And then run `svc deploy` to redeploy all the Load Balanced Web Services before `env deploy` to enable CloudFront TLS termination.
```
Comment (efekarakus) [inline site/content/docs/developing/content-delivery.en.md]:
```suggestion
    2. Run `svc deploy`individually to redeploy all the Load Balanced Web Services before enabling CloudFront TLS termination.
```

File: site/content/docs/developing/content-delivery.en.md [hunk 6]
```diff
@@ -57,20 +57,17 @@ You can restrict incoming traffic to come from a certain source. For CloudFront,
...
+    3. Once all your Load Balanced Web Services no longer redirect HTTP to HTTPS, you can safely enable CloudFront TLS termination in the env manifest.
```
Comment (efekarakus) [inline site/content/docs/developing/content-delivery.en.md]:
```suggestion
    3. Once all your Load Balanced Web Services no longer redirect HTTP to HTTPS, you can safely enable CloudFront TLS termination in the env manifest and run `env deploy`.
```

File: site/content/docs/manifest/lb-web-service.en.md [hunk 7]
```diff
@@ -259,8 +259,8 @@ http:
   hosted_zone: Z0873220N255IR3MTNR4
 # Also see http.alias array of maps example, above.
 ```
-<span class="parent-field">http.</span><a id="http-disable-redirect" href="#http-disable-redirect" class="field">`disable_redirect`</a> <span class="type">Boolean</span>  
-Disable automatic Application Load Balancer redirect from HTTP to HTTPS.
+<span class="parent-field">http.</span><a id="http-redirect-to-https" href="#http-redirect-to-https" class="field">`redirect_to_https`</a> <span class="type">Boolean</span>  
+Automatically redirect Application Load Balancer from HTTP to HTTPS.
```
Comment (efekarakus) [inline site/content/docs/manifest/lb-web-service.en.md]: Should we say the default is true?

File: site/content/docs/developing/content-delivery.en.md [hunk 8]
```diff
@@ -53,4 +53,21 @@ With both of these setups, Copilot will provision CloudFront to use the [SSL/TLS
 
 ## What is ingress restriction?
 
-You can restrict incoming traffic to come from a certain source. For CloudFront, Copilot uses an [AWS managed prefix list](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-aws-managed-prefix-lists.html) to restrict allowed traffic to a set of CIDR IP addresses associated with CloudFront edge locations. When you specify `restrict_to.cdn: true`, your Public Load Balancer is no longer fully publicly accessible, and can only be accessed through the CloudFront distribution, guarding against security threats to your services.
\ No newline at end of file
+You can restrict incoming traffic to come from a certain source. For CloudFront, Copilot uses an [AWS managed prefix list](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-aws-managed-prefix-lists.html) to restrict allowed traffic to a set of CIDR IP addresses associated with CloudFront edge locations. When you specify `restrict_to.cdn: true`, your Public Load Balancer is no longer fully publicly accessible, and can only be accessed through the CloudFront distribution, guarding against security threats to your services.
```
Comment (dannyrandall) [inline site/content/docs/developing/content-delivery.en.md]:
```suggestion
You can restrict incoming traffic to come from a certain source. For CloudFront, Copilot uses an [AWS managed prefix list](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-aws-managed-prefix-lists.html) to restrict allowed traffic to a set of CIDR IP addresses associated with CloudFront edge locations. When you specify `restrict_to.cdn: true`, your Public Load Balancer is no longer publicly accessible, and can only be accessed through the CloudFront distribution, guarding against security threats to your services.
```

File: site/content/docs/developing/content-delivery.en.md [hunk 9]
```diff
@@ -53,4 +53,21 @@ With both of these setups, Copilot will provision CloudFront to use the [SSL/TLS
...
\ No newline at end of file
+You can restrict incoming traffic to come from a certain source. For CloudFront, Copilot uses an [AWS managed prefix list](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-aws-managed-prefix-lists.html) to restrict allowed traffic to a set of CIDR IP addresses associated with CloudFront edge locations. When you specify `restrict_to.cdn: true`, your Public Load Balancer is no longer publicly accessible, and can only be accessed through the CloudFront distribution, guarding against security threats to your services.
+
+## How do I use CloudFront to terminate TLS?
+
+!!! attention
+    1. First, disable [HTTP to HTTPS redirection](../manifest/lb-web-service/#http-redirect-to-https) for your Load Balanced Web Services.
```
Comment (huanjani) [inline site/content/docs/developing/content-delivery.en.md]: nit (not necessary bc numbering, and more consistent with 2 and 3):
```suggestion
    1. Disable [HTTP to HTTPS redirection](../manifest/lb-web-service/#http-redirect-to-https) for your Load Balanced Web Services.
```

File: site/content/docs/manifest/environment.en.md [hunk 10]
```diff
@@ -247,7 +256,7 @@ http:
 ```
 
 <span class="parent-field">http.public.access_logs.</span><a id="http-public-access-logs-bucket-name" href="#http-public-access-logs-bucket-name" class="field">`bucket_name`</a> <span class="type">String</span>   
-The name of an existing S3 bucket to which to store the access logs. 
+The name of an existing S3 bucket to which to store the access logs.
```
Comment (huanjani) [inline site/content/docs/manifest/environment.en.md]: maybe?
```suggestion
The name of an existing S3 bucket in which to store the access logs.
```

File: site/content/docs/manifest/lb-web-service.en.md [hunk 11]
```diff
@@ -259,6 +259,9 @@ http:
   hosted_zone: Z0873220N255IR3MTNR4
 # Also see http.alias array of maps example, above.
 ```
+<span class="parent-field">http.</span><a id="http-redirect-to-https" href="#http-redirect-to-https" class="field">`redirect_to_https`</a> <span class="type">Boolean</span>  
+Automatically redirect Application Load Balancer from HTTP to HTTPS. By default it is `true`.
```
Comment (huanjani) [inline site/content/docs/manifest/lb-web-service.en.md]:
```suggestion
Automatically redirect the Application Load Balancer from HTTP to HTTPS. By default it is `true`.
```

**Review by efekarakus — APPROVED** [review]

**Review by dannyrandall — APPROVED** [review]

**Pull request merged**